contrib/hamilton/contrib/user/skrawcz/fine_tuning/README.md

```diff
@@ -0,0 +1,130 @@
+# Purpose of this module
+This module shows you how to fine-tune an LLM model. This code is inspired by this [fine-tuning code](https://github.com/dagster-io/dagster_llm_finetune/tree/main).
+
+Specifically the code here, from an approach standpoint, shows Supervised Fine-Tuning (SFT) for dialogue. This approach instructs the model to be more
+useful to directly respond to a question, rather than optimizing over an entire dialogue. SFT is the most common type of fine-tuning,
+as the other two options: Pre-training for Completion, and RLHF, required more to work. Pre-training requires more computational power,
+while RLHF requires higher-quality dialogue data.
+
+This code should work on a regular CPU (in a docker container), which will allow you to test out the code locally without
+any additional setup. This specific approach this code uses is [LoRA](https://arxiv.org/abs/2106.09685) (low-rank adaptation of large language models), which
+means that only a subset of the LLM's parameters are tweaked and prevents over-fitting.
```

Comment:
Add more info about the issues with mac/MPS

Reply:
I don't have much to say here though.